test/requests/code_actions_formatting_test.rb

```diff
@@ -0,0 +1,106 @@
+# typed: true
+# frozen_string_literal: true
+
+require "test_helper"
+require "expectations/expectations_test_runner"
+
+# Tests RuboCop disable directives - before/after on whole file
+class CodeActionsFormattingTest < Minitest::Test
+  TEST_EXP_DIR = File.join(ExpectationsTestRunner::TEST_EXP_DIR, "code_actions_formatting")
+  TEST_FIXTURES_DIR = ExpectationsTestRunner::TEST_FIXTURES_DIR
+
+  def test_disable_line__emoji
+    assert_disable_line("emoji", "Lint/UselessAssignment")
+  end
+
+  def test_disable_line__lambda_indentation
+    assert_disable_line("lambda_indentation", "Layout/IndentationConsistency")
+  end
+
+  def test_disable_line__comment_on_comment
+    assert_disable_line("comment_on_comment", "Layout/CommentIndentation")
+  end
+
+  def test_disable_line__multiline_array
+    assert_disable_line("multiline_array", "Layout/IndentationConsistency")
+  end
+
+  def test_disable_line__line_disable_existing
+    assert_disable_line("line_disable_existing", "Lint/UselessAssignment")
+  end
+
+  private
+
+  def assert_disable_line(fixture, cop_name)
+    assert_fixtures_match(
+      "rubocop_#{fixture}",
+      cop_name,
+      "Disable #{cop_name} for this line",
+    )
+  end
+
+  def assert_fixtures_match(name, diagnostic_code, code_action_title)
+    actual, expected = load_expectation(name)
+    assert_corrects_to_expected(
+      diagnostic_code,
+      code_action_title,
+      actual,
+      expected,
+    )
+  end
+
+  sig do
+    params(
+      diagnostic_code: String,
+      code_action_title: String,
+      source: String,
+      expected: String,
+    ).returns(T.untyped)
+  end
+  def assert_corrects_to_expected(diagnostic_code, code_action_title, source, expected)
+    document = RubyLsp::Document.new(
+      source: source.dup,
+      version: 1,
+      uri: URI::Generic.from_path(path: __FILE__),
+      encoding: LanguageServer::Protocol::Constant::PositionEncodingKind::UTF16,
+    )
+
+    diagnostics = RubyLsp::Requests::Diagnostics.new(document).run
+    diagnostic = T.must(T.must(diagnostics).find { |d| d.code == diagnostic_code })
+    range = diagnostic.range.to_hash.transform_values(&:to_hash)
+    result = RubyLsp::Requests::CodeActions.new(document, range, {
+      diagnostics: [JSON.parse(T.must(diagnostic).to_json, symbolize_names: true)],
+    }).run
+
+    # deal with weird typing in CodeActions#run lying about its type
```

Comment:
Can you explain what you mean with this? What's the type here and what error does it cause?